src/cosalib/qemu.py

```diff
@@ -0,0 +1,173 @@
+
+        # this is used in case the image has a different disk name than the platform
+        self.platform_image_name = kwargs.get("platform_image_name", self.platform)
+
+        _Build.__init__(self, *args, **kwargs)
+
+
+    def __del__(self):
+        try:
+            shutil.rmtree(self._tmpdir)
+        except Exception as e:
+            raise Exception(
+                f"failed to remove temporary directory: {self._tmpdir}", e)
+
+    @property
+    def image_qemu(self):
+        qimage = os.path.join(self.build_dir, self.meta['images']['qemu']['path'])
+        if not os.path.exists(qimage):
+            raise ImageError(f"{qimage} does not exist")
+        return qimage
+
+    @property
+    def image_name(self):
+        """
+        Get the name of the image.
+        """
+        base_name = self.build_name
+        return  f'{base_name}-{self.build_id}-{self.platform}.{BASEARCH}.{self.image_suffix}'
+
+    @property
+    def tmp_image(self):
+        tmp_image_base = os.path.basename(self.image_qemu)
+        return os.path.join(self._tmpdir, f"{tmp_image_base}.working")
+
+    @property
+    def image_path(self):
+        return os.path.join(self.build_dir, self.image_name)
+
+    @property
+    def have_artifact(self):
+        if os.path.exists(self.image_path):
+            return True
+        return False
+
+    @property
+    def image_meta(self):
+        try:
+            return self.meta["images"][self.platform]
+        except:
+            return None
+
+    def set_platform(self):
+        run_verbose(['/usr/lib/coreos-assembler/gf-platformid',
+                     self.image_qemu, self.tmp_image, self.platform])
+
+
+    def mk_image(self):
+        """
+        mk_image is broken out seperately to allow other Classes to override
+        the behavor
+        """
+        work_img = os.path.join(self._tmpdir, self.image_name)
+        final_img= os.path.join(os.path.abspath(self.build_dir), self.image_name)
+
+        self.set_platform()
+        cmd = ['qemu-img', 'convert', '-f', 'qcow2', '-O', self.image_format, self.tmp_image]
+        for k, v in self.convert_options.items():
+            cmd.extend([k, v])
+        cmd.extend([work_img])
+        run_verbose(cmd)
+        shutil.move(work_img, final_img)
+
+        fsize = '{}'.format(os.stat(self.image_path).st_size)
+        self._found_files[self.image_name] = {
+            "path": self.image_name,
+            "sha256sum": sha256sum_file(self.image_path),
+            "size": int(fsize)
+        }
+
+    def _build_artifacts(self, *args, **kwargs):
+        if self.have_artifact:
+            log.info(f"Artifact {self.image_name} exists")
+            if kwargs.get("force"):
+                log.info(f"Rebuilding artifact {self.image_name}")
+
+        self.mk_image()
+        imgs = self.meta.get("images", {})
+        imgs[self.platform] = self._found_files[self.image_name]
+        self.meta_write()
+
+
+class AWSImage(_QemuImage):
```

Comment:
I feel like these should either move out of this module, or the module should be renamed from `qemu` to something more generic.